lint/rule_target_job_instance_test.go

```diff
@@ -0,0 +1,98 @@
+package lint
+
+import (
+	"fmt"
+	"testing"
+)
+
+func testTargetRequiredMatcherRule(t *testing.T, matcher string) {
+	var linter *TargetRuleFunc
+
+	switch matcher {
+	case "job":
+		linter = NewTargetJobRule()
+	case "instance":
+		linter = NewTargetInstanceRule()
+	default:
+		t.Errorf("No concrete target required matcher rule for '%s", matcher)
+		return
+	}
```

Comment:
(Super minor nit, feel free to ignore) To avoid this I'd be tempter to pass the linter in the testcase structs below, or even stick an inner loop inside the test (will give an example).